payment_methods/Paypal_Express/ItemizedOrder.php

```diff
@@ -0,0 +1,306 @@
+        // Digital item is sold.
+        $this->order_items['L_PAYMENTREQUEST_0_ITEMCATEGORY0'] = 'Physical';
+        // Item's sales S/H and tax amount.
+        $this->order_items['PAYMENTREQUEST_0_ITEMAMT']     = $this->gateway_data_formatter->formatCurrency(
+            $this->payment->amount(),
+            $this->decimal_precision
+        );
+        $this->order_items['PAYMENTREQUEST_0_TAXAMT']      = '0';
+        $this->order_items['PAYMENTREQUEST_0_SHIPPINGAMT'] = '0';
+        $this->order_items['PAYMENTREQUEST_0_HANDLINGAMT'] = '0';
+    }
+
+
+    /**
+     * @param EE_Line_Item $total_line_item
+     * @param int          $item_num
+     * @return int
+     * @throws EE_Error
+     * @throws ReflectionException
+     */
+    private function addOrderItemsForLineItems(EE_Line_Item $total_line_item, int $item_num): int
+    {
+        // Go through each item in the list.
+        foreach ($total_line_item->get_items() as $line_item) {
+            if ($line_item instanceof EE_Line_Item) {
+                // PayPal doesn't like line items with 0.00 amount, so we may skip those.
+                if (EEH_Money::compare_floats($line_item->pretaxTotal(), '0.00', '==')) {
+                    continue;
+                }
+                $unit_price         = $line_item->unit_price();
+                $line_item_quantity = $line_item->quantity();
+                // This is a discount.
+                if ($line_item->is_percent()) {
+                    $unit_price         = $line_item->pretaxTotal();
+                    $line_item_quantity = 1;
+                }
+                // Item Name.
+                $this->order_items["L_PAYMENTREQUEST_0_NAME{$item_num}"] = mb_strcut(
+                    $this->gateway_data_formatter->formatLineItemName($line_item, $this->payment),
+                    0,
+                    127
+                );
+                // Item description.
+                $this->order_items["L_PAYMENTREQUEST_0_DESC{$item_num}"] = mb_strcut(
+                    $this->gateway_data_formatter->formatLineItemDesc($line_item, $this->payment),
+                    0,
+                    127
+                );
+                // Cost of individual item.
+                $this->order_items["L_PAYMENTREQUEST_0_AMT{$item_num}"] = $this->gateway_data_formatter->formatCurrency(
+                    $unit_price,
+                    $this->decimal_precision
+                );
+                // Item Number.
+                $this->order_items["L_PAYMENTREQUEST_0_NUMBER{$item_num}"] = $item_num + 1;
+                // Item quantity.
+                $this->order_items["L_PAYMENTREQUEST_0_QTY{$item_num}"] = $line_item_quantity;
+                // Digital item is sold.
+                $this->order_items["L_PAYMENTREQUEST_0_ITEMCATEGORY{$item_num}"] = 'Physical';
+                $this->itemized_order_sum                                        += $line_item->pretaxTotal();
```

Comment:
`$this->itemized_order_sum += $line_item->pretaxTotal();`

Reply:
thnx, will fix 👍🏻